bin/inference/pycbc_inference_plot_gelman_rubin

```diff
@@ -0,0 +1,122 @@
+""" Plots the Gelman-Rubin convergence diagnositic statistic.
+"""
+
+import argparse
+import logging
+from pycbc import results
+from pycbc.inference import gelman_rubin
+from pycbc.inference import option_utils
+import matplotlib.pyplot as plt
+import sys
+
+# command line usage
+parser = argparse.ArgumentParser(usage=__file__ + " [--options]",
+                                 description=__doc__)
+
+# verbose option
+parser.add_argument("--verbose", action="store_true", default=False,
+                    help="Print logging info.")
+
+# output options
+parser.add_argument("--output-file", type=str, required=True,
+                    help="Path to output plot.")
+parser.add_argument("--walkers", type=int, nargs="+", default=None,
+                    help="Specific walkers to plot. Default is plot "
+                         "all walkers.")
+parser.add_argument("--hline", type=float, default=None,
+                    help="Plots a horizontal line.")
+parser.add_argument("--segment-start", type=int, required=True,
+                    help="Index in chain to start calculation.")
+parser.add_argument("--segment-end", type=int, required=True,
+                    help="Index in chain to end calculation.")
+parser.add_argument("--segment-step", type=int, required=True,
+                    help="Step size in chain to next calculation.")
+
+# add results group
+option_utils.add_inference_results_option_group(parser)
+
+# parse the command line
+opts = parser.parse_args()
+
+# setup log
+if opts.verbose:
+    log_level = logging.DEBUG
+else:
+    log_level = logging.WARN
+logging.basicConfig(format="%(asctime)s : %(message)s", level=log_level)
+
+# enfore that this is not a single iteration
+if opts.iteration is not None:
+    raise ValueError("Cannot use --iteration")
+
+# load the results
+fp, params, labels, _ = option_utils.results_from_cli(
+                                         opts, load_samples=False,
+                                         walkers=None)
+
+# if use wants specific walkers
+walkers = range(fp.nwalkers) if opts.walkers is None else opts.walkers
+
+# create Figure
+fig = plt.figure()
+
+# loop over parameters
+stats = []
+for param, label in zip(params, labels):
+    logging.info("Plotting parameter %s", param)
+
+    # get samples for each chain
+    chains = [fp.read_samples(param, walkers=i, thin_start=opts.thin_start,
+                              thin_interval=opts.thin_interval,
+                              thin_end=opts.thin_end)[param] for i in walkers]
+
+    # calculate the Gelman-Rubin convergence diagnostic statistic
+    chains = [chain.reshape(1, len(chain)) for chain in chains]
+    starts, ends, stats = gelman_rubin.walk(chains, opts.segment_start,
+                                            opts.segment_end,
+                                            opts.segment_step)
+#                                chains, 50, 39999, 1000)
+
+    # plot
+    plt.plot(ends, stats[0, :], label=label)
+
```

Comment:
It would be nice to see the legend showing up on the plot. So maybe adding 
`plt.legend(loc='best', labelspacing=0.2)` would be helpful.